src/communication/network_interface_cl.c

```diff
@@ -8124,6 +8124,285 @@ logtb_dump_trantable (FILE * outfp)
 #endif /* !CS_MODE */
 }
 
+/*
+ * mmon_get_server_info -
+ *
+ * return: cubrid error
+ *
+ *   server_info(in/out): save memory usage information of the server
+ */
+int
+mmon_get_server_info (MMON_SERVER_INFO & server_info)
+{
+#if defined(CS_MODE)
+  char *buffer, *ptr, *temp_str;
+  int bufsize = 0;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int req_error, ival;
+  int error = NO_ERROR;
+
+  req_error =
+    net_client_request2 (NET_SERVER_MMON_GET_SERVER_INFO, NULL, 0, reply,
+			 OR_ALIGNED_BUF_SIZE (a_reply), NULL, 0, &buffer, &bufsize);
+  if (req_error)
+    {
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+    }
+  else
+    {
+      ptr = reply;
+      ptr = or_unpack_int (ptr, &ival);
+      ptr = or_unpack_int (ptr, &ival);
+      error = (int) ival;
+    }
+
+  if (error == NO_ERROR)
+    {
+      ptr = or_unpack_string_nocopy (buffer, &temp_str);
+      memcpy (server_info.name, temp_str, strlen (temp_str) + 1);
+      ptr = or_unpack_int64 (ptr, (int64_t *) & (server_info.total_mem_usage));
+    }
+  free_and_init (buffer);
+
+  return error;
+#else /* CS_MODE */
+  er_log_debug (ARG_FILE_LINE, "mmon_get_server_info: THIS IS ONLY a C/S function");
+  return ER_FAILED;
+#endif /* !CS_MODE */
+}
+
+/*
+ * mmon_get_module_info -
+ *
+ * return: cubrid error
+ *
+ *   module_index(in): index of module registered in memory_monitor
+ *   module_info(in/out): save memory usage information of the module
+ */
+// *INDENT-OFF*
+int
+mmon_get_module_info (int module_index, std::vector<MMON_MODULE_INFO> & module_info)
+// *INDENT-ON*
+{
+#if defined(CS_MODE)
+  char *buffer, *ptr, *temp_str;
+  int bufsize = 0;
+  OR_ALIGNED_BUF (OR_INT_SIZE) a_request;
+  char *request = OR_ALIGNED_BUF_START (a_request);
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  int req_error, ival, module_count;
+  int error = NO_ERROR;
+
+  ptr = or_pack_int (request, module_index);
+
+  req_error =
+    net_client_request2 (NET_SERVER_MMON_GET_MODULE_INFO, request, OR_ALIGNED_BUF_SIZE (a_request),
+			 reply, OR_ALIGNED_BUF_SIZE (a_reply), NULL, 0, &buffer, &bufsize);
+  if (req_error)
+    {
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+    }
+  else
+    {
+      ptr = reply;
+      ptr = or_unpack_int (ptr, &ival);
+      ptr = or_unpack_int (ptr, &ival);
```

Comment:
What does `ival` mean? Can we just ignore the previous packed values before error, which are overwritten? If it's the case, why they should be packed and sent?

Reply:
Looking around other functions using `net_client_request2`, they use the first unpacked int, area_size, to decide to proceed to unpack variables. I don't really know about it, but I doubt we can just ignore it. Could you elaborate on it if we can ignore it?